scalding-db/README.md

```diff
@@ -1,10 +1,177 @@
-## Scalding JDBC Macros
+## scalding-db
 
-Provides macros to interop between Scala case classes and relational database / SQL column definitions.
+`scalding-db` module provides Scalding sources for connecting to relational databases. A source can be created using two simple steps:
+
+Step 1 -- Defining a case class that represents your relational table schema:
+
+```scala
+case class ExampleDBRecord(
+  card_id: Long,
+  tweet_id: Long,
+  created_at: Option[java.util.Date],
+  deleted: Boolean = false)
+```
+
+Step 2 -- Defining a `TypedJDBCSource` based on the above case class:
+
+```scala
+case class ExampleDBRecordsTable(implicit dbsInEnv : AvailableDatabases)
+    extends TypedJDBCSource[ExampleDBRecord](dbsInEnv) {
+  override val tableName = TableName("example_table")
+  override val database = Database("example_schema") 
+  override val filterCondition = Some("deleted=0")
+}
+```
+
+Under the hood, the supplied case class is automatically mapped to the underlying DB schema using Scala macros.
+
+
+### Supported Mappings
+
+Scala type  | SQL type
+------------- | -------------
+`Int` | `INTEGER`
+`Long` | `BIGINT`
+`Short` | `SMALLINT`
+`Double` | `DOUBLE`
+`@varchar @size(20) String `| `VARCHAR(20)`
+`@text String` | `TEXT`
+`java.util.Date` | `DATETIME`
+`@date java.util.Date` | `DATE`
+`Boolean` | `BOOLEAN`
+          |  <sub><sup>(`BOOLEAN` is used if creating a new table at write time, but `BOOL` and `TINYINT` are also supported for reading existing columns)</sup></sub>
+
+* Annotations are used for String types to clearly distinguish between TEXT and VARCHAR column types
+* Scala `Option`s can be used to denote columns that are `NULLABLE` in the DB
+* `java.lang.*` types are not supported. For e.g. `Integer` (`java.lang.Integer`) does not work
+
+#### Working with nested case classes
+
+Nested case classes can be used as a workaround for the 22-size limitation. They can also be used for logically grouping the table columns. Nested case classes are flattened in left to right order. For example:
+```scala
+case class Person(id: Long, name: String, location: Location)
+case class Location(geo: GeoCode, doorNum: Int, street: String, city: String)
+case class GeoCode(lat: Long, lng: Long)
+```
+is flattened to a table schema with columns `id`, `name`, `lat`, `lng`, `doorNum`, `street`, `city`.
```

Comment:
what happens if you have two Persons? How are the fields assigned? Can you manually rename with annotations?